draft-ietf-quic-v2.md

```diff
@@ -178,17 +178,31 @@ server MUST include the relevant transport parameters to validate that the
 server sent the Retry and the connection IDs used in the exchange, as described
 in {{Section 7.3 of QUIC}}.
 
-The server SHOULD start sending its Initial packets using the negotiated
-version as soon as it decides to change. Before the server is able to process
-transport parameters from the client, it might need to respond to Initial
-packets from the client. For these packets the server uses the original version.
-
-Once the client has processed a packet using the negotiated version, it SHOULD
-send subsequent Initial packets using that version. The server MUST NOT discard
-its original version Initial receive keys until it successfully processes a
+The server cannot send CRYPTO frames until it has processed the client's
+transport parameters. Therefore, the server MUST send all CRYPTO frames using
+the negotiated version.
+
+The client learns the negotiated version by observing the first long header
+Version field that differs from the original version. If the client receives a
+CRYPTO frame from the server in the original version, that indicates that the
+negotiated version is equal to the original version.
+
+After learning the negotiated version, clients SHOULD discard packets with other
+versions, though they MAY decrypt Initial packets with the original version and
+process them if they fill a hole in the Initial sequence number space. Clients
+MUST discard packets with the original version if their packet number exceeds
+the lowest packet number with the negotiated version.
```

Comment:
This packet number checking doesn't make sense because we don't require the server to send all initials with the negotiated version. I'd remove this entire paragraph.